packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/flex.dart

```diff
@@ -4,204 +4,549 @@
 
 import 'dart:math';
 
+import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter/rendering.dart';
 import 'package:flutter/widgets.dart';
 
-import '../../diagnostics_node.dart';
+import '../../../ui/colors.dart';
+import '../../../ui/theme.dart';
 import '../inspector_data_models.dart';
 import 'arrow.dart';
 import 'utils.dart';
 
-@immutable
-class StoryOfYourFlexWidget extends StatelessWidget {
-  const StoryOfYourFlexWidget({
-    this.diagnostic,
-    this.properties,
+const widthIndicatorColor = mainUiColor;
+const heightIndicatorColor = mainGpuColor;
+
+String getCrossAxisAssetImageUrl(CrossAxisAlignment alignment) {
+  return 'assets/img/story_of_layout/cross_axis_alignment/${describeEnum(alignment)}.png';
+}
+
+String getMainAxisAssetImageUrl(MainAxisAlignment alignment) {
+  return 'assets/img/story_of_layout/main_axis_alignment/${describeEnum(alignment)}.png';
+}
+
+class StoryOfYourFlexWidget extends StatefulWidget {
+  const StoryOfYourFlexWidget(
+    this.properties, {
     Key key,
-  }) : super(key: key);
+  })  : assert(properties != null),
+        super(key: key);
+
+  final FlexLayoutProperties properties;
+
+  @override
+  _StoryOfYourFlexWidgetState createState() => _StoryOfYourFlexWidgetState();
+}
+
+class _StoryOfYourFlexWidgetState extends State<StoryOfYourFlexWidget> {
+  static const kBottomHeight = 16.0;
+  static const kRightWidth = 16.0;
+  static const kArrowHeadSize = 8.0;
+  static const kDistanceToArrow = 1.0;
+  static const kMargin = 8.0;
+  static const kRenderedMinWidth = 175.0;
+  static const kRenderedMinHeight = 150.0;
+  static const kDropdownMaxWidth = 325.0;
+
+  int totalFlexFactor;
+  MainAxisAlignment mainAxisAlignment;
+  CrossAxisAlignment crossAxisAlignment;
+
+  Size get size => properties.size;
+
+  FlexLayoutProperties get properties => widget.properties;
+
+  List<LayoutProperties> get children => widget.properties.children;
+
+  Axis get direction => widget.properties.direction;
 
-  final RemoteDiagnosticsNode diagnostic;
-  final RenderFlexProperties properties;
+  bool get isRow => properties.direction == Axis.horizontal;
 
-  List<Widget> _visualizeChildren(BuildContext context) {
-    if (!diagnostic.hasChildren) return [const SizedBox()];
+  bool get isColumn => !isRow;
+
+  Color get horizontalColor =>
+      properties.isHorizontalMainAxis ? mainAxisColor : crossAxisColor;
+
+  Color get verticalColor =>
+      properties.isVerticalMainAxis ? mainAxisColor : crossAxisColor;
+
+  String get flexType => properties.type.toString();
+
+  void _update() {
+    totalFlexFactor = properties.totalFlex;
```

Comment:
no need to cache these. Just add a getter that reads them off properties or cache them if you prefer.

Reply:
Initially I added this because there will be the interactive flex changer feature, but I will remove this for now.